README.md
Improved Blizzard UI (Classic Edition)
====================

Improved Blizzard UI is an attempt to improve the World of Warcraft interface by styling frames, implementing additional functionality and restructuring / hiding existing elements.

Due to the drastic differences between Classic and Mainline WoW this has been forked to it's own thing.

This is a complete ground up rewrite using Ace3 to make porting between versions a bit easier and the upgrade process between patches smoother.

![WoWScrnShot_120619_055018](https://user-images.githubusercontent.com/7526918/70299451-d74ba100-17ec-11ea-88fa-640d90643ef2.jpg)
![WoWScrnShot_120619_055244](https://user-images.githubusercontent.com/7526918/70299456-d9156480-17ec-11ea-876f-912ddc83ed88.jpg)

# Project Development Setup

To get a working installation of Improved Blizzard UI you must first clone the repository to a directory of your choosing. 

Execute `./setup.sh` - This will pull in all dependencies and put them in the libs folder.

This will require you to have the following:

- SVN CMD (Eg, [SlikSVN](https://sliksvn.com/download/)) 
- ZIP Extension (If on Windows install [Cygwin](https://www.cygwin.com/))
- Ability to open shell files (Eg, [Git BASH](https://gitforwindows.org/))

# Addon Installation

* To install Improved Blizzard UI place the folder into `World of Warcraft//_classic_//Interface//Addons` as you would any other addon.

# Features

## Customisation

* Once installed if need be customize your installation with `/imp`.
* Most UI elements can now be repositioned with `/imp unlock` and `/imp lock`.
* LibSharedMedia-3.0 Support for customising fonts. Any fonts loaded by LSM will be available in Improved Blizzard UI and vice versa. You may now easily use the Improved Blizzard UI in other addons such as Recount, DBM etc.

## Miscellaneous

* AFK 'Hero Mode' Camera View.
* Automatic Repair.
* Automatic Trash Item Sale.

## Unit Frames

### Player Frame

* Moved and Re-Scaled.
* Larger Health Bar.
* Scaleable.
* Class Coloured Health Bar.
* Hidden Portrait Text Spam.
* Hidden when out of Combat (Without Target / Low Health).
* Improved Fonts.

### Target Frame

* Moved and Re-Scaled.
* Larger Health Bar.
* Scaleable.
* Class Coloured Health Bar.
* Improved Fonts.
* Buffs on Top.

### Target of Target

* Improved Fonts.
* Scaleable.
* Class Coloured Health Bar.

### Party Frames

* Moved and Re-Scaled.
* Improved Fonts.

## Action Bars

* Casting Bar Repositioned and Scaled.
* Scaleable Buffs and Debuffs.

## Mini Map

* Moved and Re-Scaled.
* Player Co-Ordinates.
* System Performance Statistics.
* Scroll Wheel Zoom.
* Improved Fonts.

## Tooltips

* Anchored to Mouse.
* Styled Tooltips.
* Unit Hostility Border.
* Coloured Unit Guild Name, Level, Faction and Race.
* Target of Target.
* Class Coloured Health Bar and Name.
* Customizable Font Size.
* Improved Font.
* Item Rarity Border.

## Chat

* Improved Chat Font.
* Shortened Blizzard Strings (Loot, Exp Gain, Profession Levels etc).

## Combat

* Low Health Warnings (50% and 25%).
* Interrupt Announcements.
* Instanced Content Player Kill Feed.
* Highlighted Killing Blows.
* Instant Ressurection.


CHANGELOG.md
# Improved Blizzard UI (Classic Edition ) Changelog

1.0.2 - Remove Bash and Batch files that will kill Curse upload.
1.0.1 - Patch 1.13.3 ToC Bump
1.0.0 - Classic Port (Version Refresh due to Fork).
2.2.0 - German localisation (Credit @Marakuja)
2.2.0 - Pet Name Font Change
2.2.0 - Fixed an issue with the Possesion Bar placement
2.1.0 - Better Player Frame Config Check
2.1.0 - Removed unit classification from focus frame.
2.1.0 - Added timers to Target and Focus cast bars (Credit @xMarok )
2.0.4 - Tweaked position of Focus Frame and also correctly set font.
2.0.3 - Kill Artifact Bar When Leveling.
2.0.3 - Correctly get Focus Frame Unit Classification.
2.0.3 - Correctly display Focus Frame Class Colours.
2.0.2 - Added a config for hiding art
2.0.1 - Removed Dev Build Print
2.0.0 - Ground Up Rewrite. Too many changes to list.
1.9.0 - Added ptBR localisation
1.8.3 - Removed Logout and Force Quit options as they are now protected.
1.8.2 - Fixed overlapping with Order Hall bar (Credit: salindersidhu)
1.8.0 - Added Shop to MicroMenu (Credit: salindersidhu)
1.8.0 - Tweaked position of the talking head frame when the top action bar is enabled. (Credit: didehvar)
1.7.2 - Fixed an issue with the focus frame not displaying correctly on resolutions that weren't 2560x1440
1.7.1 - Removed Chat Bubble skinning since as of 7.2.0 these are protected in instances.
1.7.0 - Bumped to Patch 7.2.5.
1.7.0 - Interrupt announcements now choose channel based on grouping. (Credit: R3nd0gg)
1.7.0 - Moved Extra Action Button.
1.6.0 - German Localization updated by Marakuja
1.6.0 - Added player class colour behind PlayerFrame name section.
1.6.0 - Only actual players have class colours displayed now.
1.5.0 - Added Health Bar Colorizing (Credit: Marakuja)
1.4.0 - Moved Focus Frame
1.4.0 - Fixed chat edit box positioning when viewing the Combat Log
1.4.0 - Fixed the unclickable area next to the action bars.
1.4.0 - Added Interrupt Announcements
1.3.3 - Attempted taint fixes related to the MicroMenu.
1.3.3 - Added PvP Toggle
1.3.2 - Fixed chat font not initializing correctly on first load
1.3.1 - Fixes to make compatible with 7.1
1.3.0 - Swapped the default Class Order Hall bar with a custom replacement
1.3.0 - Fixed the /impblizz grid command
1.3.0 - Fixed a c stack overflow related to the experience bar
1.3.0 - Tweaked layout and size of config panel
1.3.0 - Styled Chat Bubbles
1.2.0 - Slightly tweaked action bar positioning
1.2.0 - Slightly tweaked party frame positioning
1.2.0 - Swapped performance counter and co-ordinates font to match the chat and bg kill feed
1.2.0 - Added a config option for hiding the chat arrows.
1.2.0 - Tweaked Honor Bar position
1.1.2 - Swapped Chat Font
1.1.2 - Fixed chat font size being reset on loading screens / relogs
1.1.1 - Fixed artifact bar hovering
1.1.0 - Removed class order hall bar. Currency can be seen under the currency tab of the player pane.
1.0.9 - Improved how class icons are detected and displayed.
Fixed minimap co-ords being generated multiple times.
Fixed display of casting bar text.
Fixed Performance Counter recreating themselves on each loading screen
AFK "Hero View" Re-Enabled
Fixed an issue with killing blows not displaying
Tweaked Buff positions due to class order halls.
Fixed a small deDE misspelling (ben2k1690)
Added Artifact bar support.
Actionbars should now adjust correctly based on reputation, exp, honor bars etc
Stancebar now adjusts position based on the presence of bottom left and right actionbars.
Added deDE localisation by ben2k1690
Minimap MicroMenu now updates based on level, unlocking features as they are required similar to the default Blizzard interface.
Honor bar moved and styled.
Slightly tweaked Battleground kill feed.
Fixed Battleground kill feed font not loading correctly.
Fixed Battleground kill feed not clearing when exiting a battleground.
Fixed a lua error when exiting vehicles.
Full rewrite
Apparently this has been printing PVEFrame open for months and months and I didn't notice. This is what happens when I unsubscribe after adding a new build.
Removed vscode folder
Made it easier to tweak action bar positon / scale.
Fixed the BG "Join Battle" button resetting to the left after every 2nd BG.
TOC bump and minor string changes.
Moved Target buffs above frame.
Nothing changed, decided to push a dev grid onto the public build as well as my own since it helps with tweaking.
Quest related crash to desktop hopefully fixed.
TOC Bump
Fixed Chat font size resetting.
Trimmed a few more global strings
Moved right end cap by 1 pixel.
Backend code change to facilitate 6.0 -> 6.1 migration
Trimmed Useless Fonts
Forced Raid Frame Class Colours.
(Hopefully) Fixed Pet Bar Taint
Chat Module Added.
Fixed PVP Group Finder window going blank after AFK mode displays.
Cleaned up localization template.
Fixed Battleground Kill Tracker Formatting Error.
Fixed pet bar issues when switching from a spec without stances to one with. Eg Disc Priest > Shadow Priest
Added cast bar timer and minor 6.1 prep.
Finally (Hopefully) fixed any pet bar positioning issues across all classes / specs.
Added Improved Blizzard UI options link to MicroMenu
Fixed incorrect TOC version.
Adjusted Boss Frame Scale and Position
Reworked Buff scaling system.
Fixed ToggleSpellbook Taint.
Minor Code Cleanup for the Cast Bar.
Stopped the config "Okay" button reloading ui even when no BlizzImp stuff has changed.
Changed Message Font (Killing Blows etc) in anticipation of Russian Localization
Forgot to remove some debug stuff
Fixed Objective Tracker Logic Bug
Scaled Boss Frames
Fixed a Blizzard bug that causes cooldowns to break when opening the fullscreen world map.
Changed Auto Repair Logic and added Guild Bank Repairs Option
Added Hide Griffin option.
Made options more pretty
Added class colours.
Added Customization Panel
Fixed bet bar and bottom left bar moving when toggling reputation bar.
Fixed Stat and Coords overlapping issues.
Fixed dk pet bar (again)
Fixed Vehicle Exit Button Position 

localisation/enGB.lua
--[[
    localisation\enGB.lua
    The English (UK) localisation of the addon.

    Essentially the default.
]]
local L = LibStub('AceLocale-3.0'):NewLocale('ImprovedBlizzardUI_Classic', 'enGB', true);

L['Miscellaneous'] = true;

-- AFK Mode
L['Enable AFK Mode'] = true;
L['After you go AFK the interface will fade away, pan your camera and display your Character in all their glory.'] = true;

-- Auto Repair
L['Auto Repair'] = true;
L['Automatically repairs your armour when you visit a merchant that can repair.'] = true;
L['Items Repaired'] = true;

-- Auto Sell
L['Auto Sell Trash'] = true;
L['Automatically sells any grey items that are in your inventory.'] = true;
L['Sold Trash Items'] = true;

-- Chat
L['Chat'] = true;
L['Chat Font'] = true;
L['Outline Font'] = true;
L['Applies a thin outline to text rendered in the chat windows.'] = true;
L['Sets the font used for the chat window, tabs etc.'] = true;
L['Style Chat'] = true;
L['Styles the Blizzard Chat frame to better match the rest of the UI.'] = true;
L['Minify Blizzard Strings'] = true;
L['Shortens chat messages such as Loot Received, Exp Gain, Skill Gain and Chat Channels.'] = true;

-- Combat
L['Combat'] = true;

-- Health Warning
L['Health Warning'] = true;
L['Health Warnings'] = true;
L['Displays a five second warning when Player Health is less than 50% and 25%.'] = true;
L['Health Warning Font'] = true;
L['The font used by the Health Warning On Screen Display Message'] = true;
L['Health Warning Size'] = true;
L['The size of the Health Warning Display.'] = true;
L['50% Colour'] = true;
L['The colour of the warning that displays at 50% health.'] = true;
L['25% Colour'] = true;
L['The colour of the warning that displays at 25% health.'] = true;
L['HP < 50% !'] = true;
L['HP < 25% !!!'] = true;

-- Interrupts
L['Interrupts'] = true;
L['Announce Interrupts'] = true;
L['When you interrupt a target your character announces this to an appropriate sound channel.'] = true;
L['Chat Channel'] = true;
L['The Channel that should be used when announcing an interrupt. Auto intelligently chooses based on situation.'] = true;
L['Interrupted X on Y'] = function (spell, target)
    return 'Interrupted '..spell..' on '..target;
end

-- Killing Blows
L['Killing Blows'] = true;
L['Highlight Killing Blows'] = true;
L['When you get a Killing Blow this will be displayed prominently in the center of the screen.'] = true;
L['Killing Blow!'] = true;
L['Killing Blow Message'] = true;
L['The message that is displayed in the center of the screen.'] = true;
L['Colour'] = true;
L['The colour of the Killing Blow notification.'] = true;
L['Killing Blow Size'] = true;
L['The size of the Killing Blow notification'] = true;
L['Killing Blow Font'] = true;
L['The font used by the Killing Blow Notification.'] = true;
L['In World'] = true;
L['Notification will display in World content.'] = true;
L['In PvP'] = true;
L['Notification will display in PvP content.'] = true;
L['In Instance'] = true;
L['Notification will display in 5 Man instanced content.'] = true;
L['In Raid'] = true;
L['Notification will display in instanced raid content.'] = true;

-- Automatic Release
L['Automatic Release'] = true;
L['Automatically release your spirit when you die.'] = true;

-- Unit Frames
L['Unit Frames'] = true;
L['Player Frame'] = true;
L['Target Frame'] = true;
L['Party Frames'] = true;

L['Style Unit Frames'] = true;
L['Applies modified textures and font styling to the Player, Target and Party Frames. This will trigger a UI Reload!'] = true;
L['Display Class Colours'] = true;
L['Hide Portrait Spam'] = true;
L['Hides the damage text that appears over the Player portrait when damaged or healed.'] = true;
L['Hide Out of Combat'] = true;
L['Hides the Player Frame when you are out of combat, have no target and are at full health.'] = true;
L['Player Frame Scale'] = true;
L['Target Frame Scale'] = true;
L['Buffs On Top'] = true;
L['Displays the Targets Buffs above the Unit Frame.'] = true;
L['ToT Class Colours'] = true;
L['Colours Target of Target Health bar to match their class.'] = true;
L['Party Frame Scale'] = true;

-- Tooltips
L['Tooltips'] = true;
L['Trivial'] = true;
L['Normal'] = true;
L['Rare'] = true;
L['Elite'] = true;
L['Rare Elite'] = true;
L['World Boss'] = true;
L['Target'] = true;
L['Anchor To Mouse'] = true;
L['The tooltip will always display at the mouse location.'] = true;
L['Style Tooltips'] = true;
L['Adjusts the information and style of the default tooltips.'] = true;
L['Guild Colour'] = true;
L['The colour of the guild name display in tooltips.'] = true;
L['Hostile Border'] = true;
L['Colours the border of the tooltip based on the hostility of the target.'] = true;
L['Class Coloured Name'] = true;
L['Colours the name of the target to match their Class.'] = true;
L['Show Target of Target'] = true;
L['Displays who / what the unit is targeting. Coloured by Class.'] = true;
L['Class Colour Health Bar'] = true;
L['Colours the Tooltip Health Bar by Class.'] = true;
L['Item Rarity Border'] = true;
L['Colours the tooltip border by the rarity of the item you are inspecting.'] = true;

-- System Stats
L['System Statistics'] = true;
L['Display System Statistics'] = true;
L['Displays FPS and Latency above the Mini Map.'] = true;
L['System Statistics Size'] = true;
L['The size of the system statistics display.'] = true;

-- Fonts
L['Primary Interface Font'] = true;
L['Replaces almost every font in the Blizzard UI to this selection. This is a broad pass.'] = true;
L['A /reload may be required after updating the primary font!'] = true;

-- Maps
L['Maps'] = true;
L['Mini Map'] = true;
L['Player Co-ordinates'] = true;
L['Co-ordinates Font'] = true;
L['The font used by the Minimap Co-ordinates Display.'] = true;
L['The colour of the Minimap Co-ordinates Display.'] = true;
L['Co-ordinates Size'] = true;
L['The size of the Minimap Co-ordinates Display.'] = true;
L['Adds a frame to the Mini Map showing the players location in the world. Does not work in Dungeons.'] = true;
L['Zone Text Font'] = true;
L['The font used by the Minimap Zone Display'] = true;
L['Zone Text Size'] = true;
L['The size of the Minimap Zone Text Display.'] = true;
L['Clock Font'] = true;
L['The font used by the Minimap Clock Display'] = true;
L['Clock Text Size'] = true;
L['The size of the Minimap Clock Display.'] = true;

L['World Map'] = true;

-- Kill Feed
L['Kill Feed'] = true;
L['Enable Kill Feed'] = true;
L['Displays a feed of the last 5 kills that occur around you.'] = true;
L['Show Casted Spell'] = true;
L['Show the Spell that caused a death.'] = true;
L['Show Damage'] = true;
L['Show how much damage the Creature or Player took.'] = true;
L['Hide When Inactive'] = true;
L['Hides the Kill Feed after no new events have occured for a short period.'] = true;
L['Kill Feed Font'] = true;
L['The font used for the Kill Feed.'] = true;
L['Text Size'] = true;
L['The font size used for the Kill Feed.'] = true;
L['Spacing'] = true;
L['The vertical spacing between each row of the Kill Feed.'] = true;
L['with'] = true;
L['Melee'] = true;
L[' killed '] = true;

-- Addon
L['Test String Output'] = true;
L['/imp - Open the configuration panel.'] = true;
L['/imp grid - Toggle a grid to aid in interface element placement.'] = true;
L['/imp unlock - Unlocks the interfaces movable frames. Locking them saves position.'] = true;
L['/imp lock - Locks the interfaces movable frames.'] = true;

L['On Screen Display'] = true;
L['Scale'] = true;
L['Font Size'] = true;

-- Action Bars
L['Bars'] = true;

-- Cast Bar
L['Cast Bar'] = true;
L['Player Cast Timer'] = true;
L['Displays a Timer on the Players Cast Bar.'] = true;

-- Buffs
L['Buffs & Debuffs'] = true;